### PR TITLE
fix linux tests and fix for linux readkey to read all keys

### PR DESCRIPTION
**Pull request by C0D3D3V**
there you go :D sorry I was stupid I forgot that it is that simple to just make the PR to your repo so you can add it to your PR ^^

### CODE REVIEW

**Comment by C0D3D3V** [comment]
would be nice if you just let the actions run completely, so we can see if the windows tests work

**Comment by Cube707** [comment]
OK, so the black precommit-hock fails, which is just a generall version problem, as far as I can tell. I will bump it

File: readchar/read_linux.py [hunk 1]
```diff
@@ -41,12 +41,16 @@ def readkey():
         return c1
 
     c2 = readchar(blocking=True)
-    if c2 != "\x5B":
+    if c2 not in ["\x4F", "\x5B"]:
```
Comment (Cube707) [inline readchar/read_linux.py:44]: please change this to `if c2 not in "\x4F\x5B":`

This keeps it shorter and is the style used on the windows side

File: readchar/read_linux.py [hunk 2]
```diff
@@ -41,12 +41,16 @@ def readkey():
...
         return c1 + c2
 
     c3 = readchar(blocking=True)
-    if c3 != "\x33":
+    if c3 not in ["\x31", "\x32", "\x33", "\x35", "\x36"]:
```
Comment (Cube707) [inline readchar/read_linux.py:48]: same here

File: readchar/read_linux.py [hunk 3]
```diff
@@ -41,12 +41,16 @@ def readkey():
...
         return c1 + c2 + c3
 
     c4 = readchar(blocking=True)
-    return c1 + c2 + c3 + c4
+    if c2 != "\x4F" or c4 not in ["\x30", "\x31", "\x33", "\x34", "\x35", "\x37", "\x38", "\x39"]:
```
Comment (Cube707) [inline readchar/read_linux.py:52]: and here

**Comment by C0D3D3V** [comment]
I can fix the to long line, or we increase the max line width :D

**Review by Cube707 — COMMENTED** [review]
reviewed your code. Please fix the commits instead of adding new ones. You can than force-push to your repo and they will be pushed to this PR.

(this way the repo styes cleaner with less unececery commits and nobody can exedentaly check out a brocken version)

File: readchar/key_windows.py [hunk 4]
```diff
@@ -33,18 +67,17 @@
...
 PAGE_DOWN = "\x00\x51"
 HOME = "\x00\x47"
 END = "\x00\x4f"
-UP = "\x00\x48"
-DOWN = "\x00\x50"
-LEFT = "\x00\x4b"
-RIGHT = "\x00\x4d"
+
+INSERT = "\x00\x52"
+SUPR = "\x00\x53"  # key.py uses SUPR not DELETE
```
[the remark this answers is not in the document]
Comment (C0D3D3V) [reply readchar/key_windows.py:83]: I already fixed this I think

**Comment by Cube707** [comment]
could you also make two commits, one for the backspace code fix on linux and one for the new windows codes

**Comment by Cube707** [comment]
also you need to change the test, so it matches the new backspace code:

https://github.com/Cube707/python-readchar/blob/d2887ef3cd001380ba2a70783026f8876beac83e/tests/linux/test_readchar.py#L20

File: tests/linux/conftest.py [hunk 5]
```diff
@@ -1,6 +1,9 @@
 import pytest
 import sys
-import select
+if sys.platform.startswith("linux"):
```
Comment (Cube707) [inline tests/linux/conftest.py:3]: black wants a blanck line before the if, thats the last thing

**Comment by Cube707** [comment]
I grabed your commits and put them on the dev-branch. They now say "authorer by you and commited by me".
You can cherry-pick them into your repo and push them to the PR to make them appear proper or I just merge the dev-branch as is and it will be pushed to the upstream PR.

However, all test work now and the code is done for now.

**Comment by C0D3D3V** [comment]
Just merge your dev branch, then you can make the git history however you like it. I have no interest in the history

**Comment by Cube707** [comment]
done. Its merged and now we continue waiting for it to get merged into the actuall project.

Have a nice remaining evening

**Pull request closed without merging**